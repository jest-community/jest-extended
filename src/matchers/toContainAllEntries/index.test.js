import matcher from './';

expect.extend(matcher);

const data = { a: 'foo', b: 'bar', c: 'baz' };

describe('.toContainAllEntries', () => {
  it('passes when object only contains all of the given entries', () => {
    expect(data).toContainAllEntries([['a', 'foo'], ['b', 'bar'], ['c', 'baz']]);
  });

  it('fails when object does not only contain all of the given entries', () => {
    expect(() => expect(data).toContainAllEntries([['a', 'foo'], ['b', 'bar']])).toThrowErrorMatchingSnapshot();
  });
});

describe('.not.toContainAllEntries', () => {
  it('passes when object does not only contain all of the given entries', () => {
    expect(data).not.toContainAllEntries([['a', 'qux'], ['b', 'bar'], ['c', 'baz']]);
  });

  it('fails when object only contains all of the given entries', () => {
    expect(() =>
      expect(data).not.toContainAllEntries([['b', 'bar'], ['a', 'foo'], ['c', 'baz']])
    ).toThrowErrorMatchingSnapshot();
  });
});
