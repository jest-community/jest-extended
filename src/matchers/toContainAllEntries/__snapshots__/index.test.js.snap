// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`.not.toContainAllEntries fails when object only contains all of the given entries 1`] = `
"expect(received).not.toContainAllEntries(expected)

Expected object to not only contain all of the given entries:
  [[\\"b\\", \\"bar\\"], [\\"a\\", \\"foo\\"], [\\"c\\", \\"baz\\"]]
Received:
  {\\"a\\": \\"foo\\", \\"b\\": \\"bar\\", \\"c\\": \\"baz\\"}"
`;

exports[`.toContainAllEntries fails when object does not only contain all of the given entries 1`] = `
"expect(received).toContainAllEntries(expected)

Expected object to only contain all of the given entries:
  [[\\"a\\", \\"foo\\"], [\\"b\\", \\"bar\\"]]
Received:
  {\\"a\\": \\"foo\\", \\"b\\": \\"bar\\", \\"c\\": \\"baz\\"}"
`;
